### PR TITLE
Zipping

### PR DESCRIPTION
**Pull request by seakayone**
Add some methods for zipping a whole project

### CODE REVIEW

File: src/main/scala/swiss/dasch/domain/Asset.scala [hunk 1]
```diff
@@ -4,24 +4,70 @@ import eu.timepit.refined.api.Refined
...
+    internalFilename: String,
+    originalInternalFilename: String,
+    originalFilename: String,
+    checksumOriginal: String,
+    checksumDerivative: String,
+  )
+
+object DotInfoFileContent {
+  implicit val decoder: JsonDecoder[DotInfoFileContent] = DeriveJsonDecoder.gen[DotInfoFileContent]
+  implicit val encoder: JsonEncoder[DotInfoFileContent] = DeriveJsonEncoder.gen[DotInfoFileContent]
```
Comment (BalduinLandolt) [inline src/main/scala/swiss/dasch/domain/Asset.scala:30]: when you have a decoder and an encoder, you can simplify that to a codec. Something along those lines:
```suggestion
  implicit val codec: JsonCodec[DotInfoFileContent] = DeriveJsonCodec.gen[DotInfoFileContent]
```

**Review by BalduinLandolt — APPROVED** [review]
LGTM

File: src/main/scala/swiss/dasch/domain/ZipUtility.scala [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    */
+  def fileInputStream(path: Path): ZIO[Scope, FileNotFoundException, FileInputStream] =
+    ZIO.fromAutoCloseable(
+      ZIO
+        .attemptBlocking(new FileInputStream(path.toFile))
+        .refineOrDie { case e: FileNotFoundException => e }
+    )
+
+  /** Creates a new managed [[ZipOutputStream]] which writes into a file using an underlying [[FileOutputStream]]. The
+    * UTF -8 charset is used to encode the entry names and comments.
```
Comment (mpro7) [inline src/main/scala/swiss/dasch/domain/ZipUtility.scala:146]:
```suggestion
    * UTF-8 charset is used to encode the entry names and comments.
```

**Review by mpro7 — APPROVED** [review]
All good, but license headers are missing.

File: src/main/scala/swiss/dasch/domain/Asset.scala (hunk 1 above)
Comment (seakayone) [reply src/main/scala/swiss/dasch/domain/Asset.scala:30]: Thanks for the hint!

**Pull request merged**